From 0141368bee1615b6a52b19319965702db200af98 Mon Sep 17 00:00:00 2001
From: "Joshua M. Clulow" <jmc@joyent.com>
Date: Mon, 11 Sep 2017 13:40:56 +0000
Subject: [PATCH] MANATEE-357 RST flood protection turns active failures into
 timeouts when PostgreSQL not running Reviewed by: Robert Mustacchi
 <rm@joyent.com> Approved by: David Pacheco <dap@joyent.com>

---
 boot/configure.sh | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/boot/configure.sh b/boot/configure.sh
index 7460723..483cb66 100755
--- a/boot/configure.sh
+++ b/boot/configure.sh
@@ -1,5 +1,4 @@
 #!/bin/bash
-# -*- mode: shell-script; fill-column: 80; -*-
 #
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
@@ -7,10 +6,23 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
-export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+export PS4
 set -o xtrace
 
+#
+# Disable the protection against RST reflection denial-of-service attacks.
+# In order for system liveliness when PostgreSQL is not running, we need to
+# be able to send a RST for every inbound connection to a closed port.  This
+# is only safe because we run Manatee on an isolated network.
+#
+# The long-term stability of this interface is not completely clear, so we
+# ignore the exit status of ndd(1M).  To do otherwise may unintentionally
+# create a flag day with future platform versions.
+#
+/usr/sbin/ndd -set /dev/tcp tcp_rst_sent_rate_enabled 0
+
 exit 0
-- 
2.21.0

